services/pds/index.js

```diff
@@ -12,135 +12,60 @@ require('dd-trace') // Only works with commonjs
 
 // Tracer code above must come before anything else
 const path = require('path')
-const { KmsKeypair, S3BlobStore } = require('@atproto/aws')
 const {
-  Database,
-  ServerConfig,
   PDS,
-  ViewMaintainer,
+  Database,
+  envToCfg,
+  envToSecrets,
+  readEnv,
+  httpLogger,
   PeriodicModerationActionReversal,
 } = require('@atproto/pds')
-const { Secp256k1Keypair } = require('@atproto/crypto')
+const pkg = require('@atproto/pds/package.json')
 
 const main = async () => {
-  const env = getEnv()
-  // Migrate using credentialed user
-  const migrateDb = Database.postgres({
-    url: pgUrl(env.dbMigrateCreds),
-    schema: env.dbSchema,
-    // We need one connection for the
-    // view-maintainer lock then one for anything else.
-    poolSize: 2,
-  })
-  // Use lower-credentialed user to run the app
-  const db = Database.postgres({
-    url: pgUrl(env.dbCreds),
-    schema: env.dbSchema,
-    poolSize: env.dbPoolSize,
-    poolMaxUses: env.dbPoolMaxUses,
-    poolIdleTimeoutMs: env.dbPoolIdleTimeoutMs,
-  })
-  const s3Blobstore = new S3BlobStore({ bucket: env.s3Bucket })
-  const repoSigningKey = await Secp256k1Keypair.import(env.repoSigningKey)
-  const plcRotationKey = await KmsKeypair.load({
-    keyId: env.plcRotationKeyId,
-  })
-  let recoveryKey
-  if (env.recoveryKeyId.startsWith('did:')) {
-    recoveryKey = env.recoveryKeyId
-  } else {
-    const recoveryKeypair = await KmsKeypair.load({
-      keyId: env.recoveryKeyId,
+  const env = readEnv()
+  env.version ??= pkg.version
+  const cfg = envToCfg(env)
+  const secrets = envToSecrets(env)
+  const pds = await PDS.create(cfg, secrets)
+  if (cfg.db.dialect === 'pg') {
+    // Migrate using credentialed user
+    const migrateDb = Database.postgres({
+      url: cfg.db.migrationUrl,
+      schema: cfg.db.schema,
     })
-    recoveryKey = recoveryKeypair.did()
+    await migrateDb.migrateToLatestOrThrow()
```

Comment:
We may not want to run migrations automatically.  Should we keep a separate service entrypoint, or maybe introduce a new config for it?

Reply:
Yup good call - i removed it 👍